magpy/gui/magpy_gui.py

```diff
@@ -5131,6 +5140,53 @@ def onLoadDI(self,event):
             self.menu_p.abs_page.AnalyzeButton.Enable()
         dlg.Destroy()
 
+    def onLoadUSGS(self,event):
+        startdate = self.plotstream.ndarray[KEYLIST.index('time')][0]
+        enddate = self.plotstream.ndarray[KEYLIST.index('time')][-1]
+        starttime = num2date(startdate)
+        endtime = num2date(enddate)
+        di_db = []
+        dlg = LoadUSGSDialog(None, title='Get USGS data')
+        if dlg.ShowModal() == wx.ID_OK:
+            extension = dlg.extensionComboBox.GetValue()
```

Comment:
Add a method to the LoadUSGSDialog that handles the logic of calculating the extension period, so onLoadUSGS doesn't need to know about the inner workings of the dialog controls...